packages/transaction-controller/src/utils/history.test.ts

```diff
@@ -0,0 +1,372 @@
+import { toHex } from '@metamask/controller-utils';
+import { add0x } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+import {
+  type TransactionHistory,
+  TransactionStatus,
+  type TransactionMeta,
+  type TransactionHistoryEntry,
+} from '../types';
+import { MAX_HISTORY_LENGTH, updateTransactionHistory } from './history';
+
+describe('History', () => {
```

Comment:
We currently test the history at the controller level in `TransactionController.test.ts` but this definitely feels warranted now given the extra logic.